src/gui_common/CustomRichTextLabel.cs

```diff
@@ -225,32 +225,52 @@ private string BuildTemplateForTag(string input, ThriveBbCode bbcode, List<strin
         // Defaults to input so if something fails output returns unchanged
         var output = input;
 
+        var simulationParameters = SimulationParameters.Instance;
+
         switch (bbcode)
         {
             case ThriveBbCode.Compound:
             {
-                if (!SimulationParameters.Instance.DoesCompoundExist(input))
-                {
-                    GD.Print($"Compound: \"{input}\" doesn't exist, referenced in bbcode");
-                    break;
-                }
-
-                var compound = SimulationParameters.Instance.GetCompound(input);
-
-                var name = compound.Name;
-
                 var pairs = StringUtils.ParseKeyValuePairs(attributes);
 
-                // Parse attributes if there are any
-                if (pairs.TryGetValue("text", out string value))
+                var fallback = false;
+                var internalName = string.Empty;
+
+                if (pairs.TryGetValue("type", out string value))
                 {
                     if (!value.StartsAndEndsWith("\""))
                         break;
 
-                    name = value.Substring(1, value.Length - 2);
+                    internalName = value.Substring(1, value.Length - 2);
+                }
+
+                if (!string.IsNullOrEmpty(internalName) && !simulationParameters.DoesCompoundExist(internalName))
```

Comment:
Wouldn't this if-else be simpler to understand if this was structured like this:
```
if(!string.IsNullOrEmpty(internalName){
   // Explicitly defined 
   setup internalName here
} else if(!string.IsNullOrEmpty(input)){
   internalName = input;
} else {
  error
}
// Then later
if(!simulationParameters.DoesCompoundExist(internalName)
  report error
```
That way the logic I think would be a bit simpler to understand and reporting the compound not existing would be unified.

Reply:
If the internal name is already explicitly defined in the type attribute, there will be nothing left to setup in the first if block as the actual processing would be after the compound validation. I think this could be coalesced further while preserving the unified invalid compound reporting into
```
// get internalname

if (!string.IsNullOrEmpty(input) && string.IsNullOrEmpty(internalName))
{
    // Handle fallback
}

if(!simulationParameters.DoesCompoundExist(internalName)
  report error and abort

// final step
```

Reply:
If that does also result in non-duplicate string emptiness checks, I'm fine with it.

Reply:
For the fallback handling at least there is no duplicate string empty checks. In the final step there's one check for empty input handling `[tag][/tag]` and no more than that.